docs/sql/statements/alter-role.rst

```diff
@@ -18,16 +18,20 @@ Synopsis
 ::
 
     ALTER ROLE name
-      SET ( parameter = value [, ...] )
+        SET ( parameter = value [, ...] )
+      | RESET [parameter | ALL]
 
 
 Description
 ===========
 
-``ALTER ROLE`` applies a change to an existing database user or role. Only
+``ALTER ROLE SET`` applies a change to an existing database user or role. Only
 existing superusers or the user itself have the privilege to alter an existing
 database user.
```

Comment:
> I don't think so

Thanks for elaborating. Could you explain why this functionality should only be accessible to superusers with direct "physical" server access, instead of admin users who can create and manage user accounts, set privileges, and configure sensible defaults for users, including accounts used for integration with other services without direct user access? Is this restriction because it makes the feature less useful for cloud environments?

Reply:
If users with `AL` can just set session settings to users, wanna help with the current situation?

Reply:
There should be a way for an `AL` user to change the defaults for other users. Imagine a user created solely for visualization/query purposes, where you want to set a `statement_timeout` without having access to the user or that user's password. As an `AL` user, you could delete that user, recreate it, log in, change the setting, share the credentials, and ask the end user to change them again, but this is a big inconvenience. For some users, a central default for all users, as described in 16315, might be an alternative, but this would not solve the issue of centrally administering other users.

While this might be possible as a superuser, in many managed or distributed environments, accessing CrateDB with superuser privileges is only possible with access to the machine, which is often managed by others rather than CrateDB itself. CrateDB's and PostgreSQL's privilege systems are still quite different, especially in terms of administrative privileges, with the limitation that the superuser is basically only safely accessible from the host machine itself, while in PostgreSQL any user can be granted `SUPERUSER` privileges similar to `AL`.

Reply:
Sorry for my short comment. The reason for that was that I was busy but didn't want to have this PR derailed in that direction.

And the reason why I consider this derailment is because we've had recently a conversation where we questioned the `AL` permission in general a bit, and were inclined to move towards supporting `grant crate to...` instead to give users full superuser privileges.

So:

- There will be some follow up to this
- Not clear yet if it will be `AL`, or if we immediately move to `grant crate to`

> Is this restriction because it makes the feature less useful for cloud environments?

We can do without baseless accusations like that, no?

Reply:
> The reason for that was that I was busy but didn't want to have this PR derailed in that direction.

I am not quite following why this was closed then: https://github.com/crate/crate/issues/16379 . It is just tracking the missing aspect or is there another open ticket that I missed?

> And the reason why I consider this derailment is because we've had recently a conversation where we questioned the AL permission in general a bit, and were inclined to move towards supporting grant crate to... instead to give users full superuser privileges.

I understand this, however I also see this as a bigger - potentially breaking change - with lots of unknowns right now. Also it doesn't align with PostgreSQL either, which does allow editing the defaults for other users with `ADMIN` privileges:

> https://www.postgresql.org/docs/current/sql-alterrole.html
> Roles having CREATEROLE privilege can change defaults for non-superuser roles for which they have been granted ADMIN OPTION.

that imho does align very much with our `AL` role.

> We can do without baseless accusations like that, no?

Sure, let's get back to collaborating on a productive level. Sorry.